components/server/src/server.rs

```diff
@@ -1843,45 +1853,52 @@ impl<EK: KvEngine, R: RaftEngine> EngineMetricsManager<EK, R> {
     }
 }
 
-pub struct EnginesResourceInfo {
-    tablet_factory: Arc<KvEngineFactory>,
+pub struct EnginesResourceInfo<T>
+where
+    T: TabletFactory<RocksEngine> + TabletAccessor<RocksEngine> + Sync + Send,
+{
+    tablet_factory: Arc<T>,
```

Comment:
Why not `Arc<dyn trait>`?

Reply:
What's the benefit of Arc<dyn trait> over this generic?

Reply:
`Arc` is used in both versions. I guess your question is why use trait object instead of generic. Just to make code consistent as `Arc<dyn TabletFactory>` is commonly used in the codebase. And use trait object should also make code simpler.